runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -353,11 +387,21 @@ impl<Cfg: Config> API for Module<Cfg> {
         Self::encode_evm_result(ctx, evm_result, tx_metadata)
     }
 
-    fn get_storage<C: Context>(_ctx: &C, address: H160, index: H256) -> Result<Vec<u8>, Error> {
-        state::with_public_storage(&address, |store| {
-            let result: H256 = store.get(index).unwrap_or_default();
-            Ok(result.as_bytes().to_vec())
-        })
+    fn get_storage<C: Context>(ctx: &C, address: H160, index: H256) -> Result<Vec<u8>, Error> {
+        let whitelisted_magic_slots = Cfg::whitelisted_magic_slots();
+        // If whitelisted magic slot, use confidential storage.
```

Comment:
You could return the result here, removing a few lines.

Reply:
In this case, I prefer the if/else instead of early returning since it avoids implying that one branch is special or short-circuiting, and makes it clearer that both branches are equal alternatives based on the condition.

But I did save a line by inlining the `Cfg::whitelisted_magic_slots` call :).

Reply:
Updated, thx!